Pinta.Core/Extensions/OtherExtensions.cs

```diff
@@ -65,41 +65,59 @@ public static PointI[][] CreatePolygonSet (this BitMask stencil, RectangleD boun
 
 				PointI last = new (start.X, start.Y + 1);
 				PointI curr = new (start.X, start.Y);
-				PointI next = curr;
-				PointI left = new ();
-				PointI right = new ();
-
+				
 				// trace island outline
 				while (true) {
-					left.X = ((curr.X - last.X) + (curr.Y - last.Y) + 2) / 2 + curr.X - 1;
-					left.Y = ((curr.Y - last.Y) - (curr.X - last.X) + 2) / 2 + curr.Y - 1;
 
-					right.X = ((curr.X - last.X) - (curr.Y - last.Y) + 2) / 2 + curr.X - 1;
-					right.Y = ((curr.Y - last.Y) + (curr.X - last.X) + 2) / 2 + curr.Y - 1;
+					// Calculation zone
+
+					int currLastDiffX = curr.X - last.X;
+					int currLastDiffY = curr.Y - last.Y;
+					int currXDecreased = curr.X - 1;
+					int currYDecreased = curr.Y - 1;
+
+					int leftX = currXDecreased + ((currLastDiffX + currLastDiffY + 2) / 2);
+					int leftY = currYDecreased + ((currLastDiffY - currLastDiffX + 2) / 2);
+
+					int rightX = currXDecreased + ((currLastDiffX - currLastDiffY + 2) / 2);
+					int rightY = currYDecreased + ((currLastDiffY + currLastDiffX + 2) / 2);
 
-					if (bounds.ContainsPoint (left.X, left.Y) && stencil[left]) {
+					PointI left = new (leftX, leftY);
+					PointI right = new (rightX, rightY);
+
+					int nextXAddition;
+					int nextYAddition;
+					if (bounds.ContainsPoint (leftX, leftY) && stencil[left]) {
```

Comment:
Having both `leftX` / `leftY` and `left` being used is a bit confusing. I'd prefer consolidating that into just `left`, `left.X` etc

A lot of these duplicate X/Y operations could maybe also be simplified by having more of the code work with `PointI` instead of separate ints (perhaps adding any necessary methods / operators into `PointI`)